inference-engine/thirdparty/movidius/mvnc/src/mvnc_api.c

```diff
@@ -82,8 +82,9 @@ static OVERLAPPED global_lock_overlap = { 0 };
 #define GLOBAL_UNLOCK() UnlockFileEx(global_lock_fd, 0, MAXDWORD, MAXDWORD, &global_lock_overlap)
 #else
 static int global_lock_fd = -1;
-#define GLOBAL_LOCK() flock(global_lock_fd, LOCK_EX)
-#define GLOBAL_UNLOCK() flock(global_lock_fd, LOCK_UN)
+#define SAFETY_CHECK(_func) do { if((_func) != 0) { mvLog(MVLOG_ERROR, "does not finish correctly"); return NC_ERROR;}} while(0)
+#define GLOBAL_LOCK() SAFETY_CHECK(flock(global_lock_fd, LOCK_EX)); SAFETY_CHECK(pthread_mutex_lock(&deviceOpenMutex))
+#define GLOBAL_UNLOCK() SAFETY_CHECK(flock(global_lock_fd, LOCK_UN)); SAFETY_CHECK(pthread_mutex_unlock(&deviceOpenMutex))
```

Comment:
In case of an error in the first call, you return NC_ERROR immediately, but you should unlock `deviceOpenMutex`

BTW, take a look at `CHECK_MUTEX_SUCCESS` and `CHECK_MUTEX_SUCCESS_RC` macros